samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,274 @@
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var signInLink = await GetSignInLinkAsync(turnContext, cancellationToken).ConfigureAwait(false);
+            if (turnContext.Activity.Text.Contains("login"))
+            {
+                string[] path = { ".", "Resources", "options.json" };
+                var member = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                var initialAdaptiveCard = GetAdaptiveCardWithAuthOptions(path, signInLink, turnContext.Activity.From.Name, member.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(initialAdaptiveCard), cancellationToken);
+            }
+            else if (turnContext.Activity.Text.Contains("PerformSSO"))
+            {
+                string[] striPath = { ".", "Resources", "AdaptiveCardWithSSOInRefresh.json" };
+                var varMember = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                var varInitialAdaptiveCard = GetAdaptiveCardWithAuthOptions(striPath, signInLink, turnContext.Activity.From.Name, varMember.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(varInitialAdaptiveCard), cancellationToken);
+            }
+            else
+            {
+                await turnContext.SendActivityAsync(MessageFactory.Text("Please send 'login' for options"), cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// The OAuth Prompt needs to see the Invoke Activity in order to complete the login process.
+        /// </summary>
+        /// <param name="turnContext">The context for the current turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "adaptiveCard/action")
+            {
+                if (turnContext.Activity.Value == null)
+                    return null;
+
+                JObject value = JsonConvert.DeserializeObject<JObject>(turnContext.Activity.Value.ToString());
+
+                if (value["action"] == null)
+                    return null;
+
+                JObject actiondata = JsonConvert.DeserializeObject<JObject>(value["action"].ToString());
+
+                if (actiondata["verb"] == null)
+                    return null;
+
+                //When adaptiveCard/action invoke activity from teams contains sso token
```

Comment:
update the comment to :

" When adaptiveCard/action invoke activity from teams contains token in response to sso flow from earlier invoke."

Reply:
Fixed